cmake/alpakaCommon.cmake

```diff
@@ -0,0 +1,881 @@
+if(ALPAKA_ACC_GPU_HIP_ONLY_MODE AND NOT ALPAKA_ACC_GPU_HIP_ENABLE)
+    message(WARNING "If ALPAKA_ACC_GPU_HIP_ONLY_MODE is enabled, ALPAKA_ACC_GPU_HIP_ENABLE has to be enabled as well.")
+    set(_ALPAKA_FOUND FALSE)
+endif()
+
+option(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE "Enable the serial CPU back-end" ${ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE "Enable the threads CPU block thread back-end" ${ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE "Enable the fibers CPU block thread back-end" ${ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE "Enable the TBB CPU grid block back-end" ${ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE "Enable the OpenMP 2.0 CPU grid block back-end" ${ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE "Enable the OpenMP 2.0 CPU block thread back-end" ${ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE_DEFAULT})
+option(ALPAKA_ACC_CPU_BT_OMP4_ENABLE "Enable the OpenMP 4.0 CPU block and block thread back-end" ${ALPAKA_ACC_CPU_BT_OMP4_ENABLE_DEFAULT})
+
+if((ALPAKA_ACC_GPU_CUDA_ONLY_MODE OR ALPAKA_ACC_GPU_HIP_ONLY_MODE)
+   AND
+    (ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE OR
+    ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE OR
+    ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE OR
+    ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE OR
+    ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR
+    ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE OR
+    ALPAKA_ACC_CPU_BT_OMP4_ENABLE))
+    if(ALPAKA_ACC_GPU_CUDA_ONLY_MODE)
+        message(WARNING "If ALPAKA_ACC_GPU_CUDA_ONLY_MODE is enabled, only back-ends using CUDA can be enabled! This allows to mix alpaka code with native CUDA code. However, this prevents any non-CUDA back-ends from being enabled.")
+    endif()
+    if(ALPAKA_ACC_GPU_HIP_ONLY_MODE)
+        message(WARNING "If ALPAKA_ACC_GPU_HIP_ONLY_MODE is enabled, only back-ends using HIP can be enabled!")
+    endif()
+    set(_ALPAKA_FOUND FALSE)
+endif()
+
+# avoids CUDA+HIP conflict
+if(ALPAKA_ACC_GPU_HIP_ENABLE AND ALPAKA_ACC_GPU_CUDA_ENABLE)
+    message(FATAL_ERROR "CUDA and HIP can not be enabled both at the same time.")
+endif()
+
+# HIP is only supported on Linux
+if(ALPAKA_ACC_GPU_HIP_ENABLE AND (MSVC OR WIN32))
+    message(WARNING "Optional alpaka dependency HIP can not be built on Windows! HIP back-end disabled!")
+    set(ALPAKA_ACC_GPU_HIP_ENABLE OFF CACHE BOOL "Enable the HIP GPU back-end" FORCE)
+endif()
+
+# Drop-down combo box in cmake-gui.
+set(ALPAKA_DEBUG "0" CACHE STRING "Debug level")
+set_property(CACHE ALPAKA_DEBUG PROPERTY STRINGS "0;1;2")
+
+set(ALPAKA_CXX_STANDARD "14" CACHE STRING "C++ standard version")
+set_property(CACHE ALPAKA_CXX_STANDARD PROPERTY STRINGS "14;17;20")
+
+if(NOT TARGET alpaka)
+    add_library(alpaka INTERFACE)
+    
+    target_compile_features(alpaka INTERFACE cxx_std_${ALPAKA_CXX_STANDARD})
+
+    add_library(alpaka::alpaka ALIAS alpaka)
```

Comment:
excellent, this means `add_subdirectory` workflows as described here will work downstream:
https://github.com/openPMD/openPMD-api#linking-to-your-project